scripts/native_libraries/get_emba_db.py

```diff
@@ -0,0 +1,86 @@
+import json
+import os
+
+import requests
+
+from surfactant.configmanager import ConfigManager
+
+
+def load_database(url):
+    response = requests.get(url)
+    response.raise_for_status()
+    return response.text
+
+
+def parse_cfg_file(content):
+    database = {}
+    lines = content.splitlines()
+    filtered_lines = []
+
+    for line in lines:
+        if not (line.startswith("#") or line.startswith("identifier")):
+            filtered_lines.append(line)
+
+    for line in filtered_lines:
+        line = line.strip()
+
+        # Split by semicolons
+        fields = line.split(";")
+
+        # Name of library
+        lib_name = fields[0]
+
+        # Empty filename because EMBA doesn't need filename patterns
+        name_patterns = []
+
+        # Check if it starts with one double quote and ends with two double quotes
+        if fields[3].startswith('"') and fields[3].endswith('""'):
+            filecontent = fields[3][1:-1]
+        elif fields[3].endswith('""'):
+            filecontent = fields[3][:-1]
+        else:
+            filecontent = fields[3].strip('"')
+
+        # Create a dictionary for this entry and add it to the database
+        if fields[1] == "" or fields[1] == "strict":
+            if fields[1] == "strict":
+                if lib_name not in database:
+                    database[lib_name] = {
+                        "filename": [lib_name],
+                        "filecontent": [],
+                    }
+            else:
+                if lib_name not in database:
+                    database[lib_name] = {
+                        "filename": name_patterns,
+                        "filecontent": [filecontent],
+                    }
+                else:
+                    database[lib_name]["filecontent"].append(filecontent)
```

Comment:
```suggestion
                try:
                    if lib_name not in database:
                        database[lib_name] = {
                            "filename": name_patterns,
                            "filecontent": [filecontent],
                        }
                    else:
                        database[lib_name]["filecontent"].append(filecontent)
                except re.error as e:
                    print(f"Error parsing file content regexp {filecontent}: {e}")
```

When constructing the database, let's catch the patterns that cause issues to make sure when we go to generate the SBOM we don't run into invalid regexp patterns.

Reply:
We could consider doing something similar for file name patterns.

Reply:
Hmm I think im crazy cause both of the above patterns compile for me. I tested it using `re.compile` as well. I'll rerun again tomorrow!

Also, regarding doing the `re` check for the file name patterns, I think in this case it won't be necessary? Cause EMBA does not have file name patterns, and if we see that `fields[1] == "strict"` then we just set the `lib_name` under the file name pattern section. So, we wouldn't have any real patterns in the file name patterns since the pattern would just be a string literal of the file name.

Reply:
Try using an older version of Python (on LC cluster such as dane.llnl.gov, `module spider python` to list available Python versions then `module spider python/3.9.12` to "load" the different version). What I'm finding is that Python <=3.10 gives the multiple repeat error, 3.11 outputs nothing, and 3.12 prints a SyntaxWarning about `\.` being an invalid escape sequence.

Since users are going to be getting the database file themselves, presumably with the same version of Python version that they will run Surfactant with, the check should catch patterns that break on their Python version. 

---

In match by attribute we're still using re.search to match the file name using a regular expression and characters such as + and ? are valid in file names, so there could be a file name that fails to compile with the `re` module:

```python
if attribute == "filename":
    matches = re.search(pattern, content)
```

If the file names in the database will always be exact matches rather than patterns, we can just do a regular string comparisons instead of pattern matching.

Reply:
Okay for filename I switched it to do a regular string comparison. Content is basically filename below:
`if name == content:`
    `libs.append({"isLibrary": name})`
    
EMBA db has about 806 libraries total, and only ~37 have the "strict" keyword. These are the ones where we set the `lib_name` to filename since EMBA does not have file content patterns for the `strict` keyword. And the filenames don't have any special characters so simple string comparison should work!